google_guest_agent/network/systemd_networkd.go

```diff
@@ -0,0 +1,112 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/run"
+	"github.com/GoogleCloudPlatform/guest-agent/metadata"
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+)
+
+const configDir = "/etc/systemd/network"
+
+type systemdnetworkd struct{}
+
+func (n systemdnetworkd) IsManaging(ctx context.Context, iface string) (bool, error) {
```

Comment:
Please add comments/doc to this function and describe the strategy we are following.